### PR TITLE
Support:  pattern match and import API for GoogleDrive and Amazon S3.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Added below APIs

- searchAndGetCSVFilesFromGoogleDrive
- searchAndGetCSVFilesFromS3
- searchAndGetExcelFilesFromS3
- searchAndGetExcelFilesFromGoogleDrive
- searchAndReadDelimFiles
- searchAndReadExcelFiles

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/aws_s3.R [hunk 1]
```diff
@@ -120,13 +120,51 @@ getCSVFilesFromS3 <- function(files, region, username, password, bucket, fileNam
...
+                              col_names = TRUE, col_types = readr::cols(.default = readr::col_character()),
+                              locale = readr::default_locale(),
+                              na = c("", "NA"), quoted_na = TRUE,
+                              comment = "", trim_ws = FALSE,
+                              skip = 0, n_max = Inf, guess_max = min(1000, n_max),
+                              progress = interactive()) {
+
+  files <- aws.s3::get_bucket_df(region = region, bucket = bucket, key = username, secret = password, max= Inf) %>%
+    filter(str_detect(Key, stringr::str_c("(?i)", searchKeyword)))
+  # set name to the files so that it can be used for the "id" column created by purrr:map_dfr.
```
Comment (hideaki) [inline R/aws_s3.R:140]: Comment may be irrelevant.

File: R/aws_s3.R [hunk 2]
```diff
@@ -120,13 +120,51 @@ getCSVFilesFromS3 <- function(files, region, username, password, bucket, fileNam
...
+                              escape_backslash = FALSE, escape_double = TRUE,
+                              col_names = TRUE, col_types = readr::cols(.default = readr::col_character()),
+                              locale = readr::default_locale(),
+                              na = c("", "NA"), quoted_na = TRUE,
+                              comment = "", trim_ws = FALSE,
+                              skip = 0, n_max = Inf, guess_max = min(1000, n_max),
+                              progress = interactive()) {
+
+  files <- aws.s3::get_bucket_df(region = region, bucket = bucket, key = username, secret = password, max= Inf) %>%
+    filter(str_detect(Key, stringr::str_c("(?i)", searchKeyword)))
```
Comment (hideaki) [inline R/aws_s3.R:139]: Add comment on regular expression.

File: R/aws_s3.R (hunk 1 above)
Comment (hidekoji) [reply R/aws_s3.R]: Removed not used comment.

File: R/aws_s3.R (hunk 2 above)
Comment (hidekoji) [reply R/aws_s3.R:140]: Added comment.

**Pull request merged**